awx/job_template_launch_resource.go

```diff
@@ -76,18 +78,29 @@ func (r *JobTemplateLaunchGetResponse) JobTemplateData() *JobTemplate {
 type JobTemplateLaunchPostRequest struct {
 	Request
 
-	extraVars string
+	extraVars map[string]interface{}
 }
 
-func (r *JobTemplateLaunchPostRequest) ExtraVars(value string) *JobTemplateLaunchPostRequest {
+func (r *JobTemplateLaunchPostRequest) ExtraVars(value map[string]interface{}) *JobTemplateLaunchPostRequest {
 	r.extraVars = value
 	return r
 }
 
+func (r *JobTemplateLaunchPostRequest) ExtraVar(name string, value interface{}) *JobTemplateLaunchPostRequest {
+	r.extraVars[name] = value
+	return r
+}
+
 func (r *JobTemplateLaunchPostRequest) Send() (response *JobTemplateLaunchPostResponse, err error) {
 	// Generate the input data:
 	input := new(data.JobTemplateLaunchPostRequest)
-	input.ExtraVars = r.extraVars
+
+	// convert extravars json to string
+	bytes, err := json.Marshal(r.extraVars)
```

Comment:
Does `json.Marshal` work well if `r.extraVars` is `nil`?

Reply:
Actually I think we do *not* need to convert this to JSON, because it will already be converted to JSON when the request is serialized. What AWX expects, if I understood correctly, is a JSON object, not a string.

Reply:
`json.Marshal` outputs `"null"` string which is not what we want. I am fixing it on the next version of the PR. nice catch